logstash-core/lib/logstash/util/substitution_variables.rb

```diff
@@ -76,6 +79,19 @@ def replace_placeholders(value)
       end
       replacement.to_s
     end
+
+    # no further action need if substitution didn't happen
+    return placeholder_value unless is_placeholder_found
+
+    # ENV ${var} value may carry single quote or escaped double quote
+    # or single/double quoted entries in array string, needs to be refined
+    refined_value = placeholder_value.gsub(/[\\"\\']/, '')
```

Comment:
I'd suggest trimming whitespace before and after the brackets, so that `start_with` and `end_with` can work with strings like `' ["a", "b", "c"] '`:

```suggestion
    refined_value = placeholder_value.strip.gsub(/[\\"\\']/, '')
```

Reply:
We have a [test case in config-mixin](https://github.com/elastic/logstash/blob/main/logstash-core/spec/logstash/config/mixin_spec.rb#L536) where it expects not to strip. We need to update test case (wonder what side affects would be) if this will be a strict requirement.